krossbow-websocket-core/src/nativeDarwinMain/kotlin/org/hildan/krossbow/websocket/darwin/DarwinWebSocketClient.kt

```diff
@@ -248,9 +248,32 @@ private fun NSData.toByteArray(): ByteArray {
     }
 }
 
-private fun NSError.toIosWebSocketException() = DarwinWebSocketException(this)
+private fun NSError.toIosWebSocketException(
+    urlResponse: NSURLResponse? = null
+): DarwinWebSocketException {
 
+    return if (urlResponse != null) {
+        val httpResponse = urlResponse as? NSHTTPURLResponse
+        val statusCode = httpResponse?.statusCode?.toInt()
+        DarwinWebSocketException(this, statusCode)
+    } else {
+        DarwinWebSocketException(this, null)
+    }
+}
 /**
  * A [WebSocketException] caused by a darwin [NSError]. It contains details about the actual error cause.
  */
-class DarwinWebSocketException(val nsError: NSError) : WebSocketException(nsError.localizedDescription)
+class DarwinWebSocketException(
+    nsError: NSError,
+    httpStatusCode: Int?,
+) : WebSocketException(exceptionMessage(nsError, httpStatusCode))
+
+private fun exceptionMessage(nsError: NSError, httpStatusCode: Int?): String {
+    val baseMessage =
+        "${nsError.domain} ${nsError.code}: ${nsError.localizedDescription}"
```

Comment:
Now I'm wondering whether `NSError` already has a string property that looks like this. Maybe a `.toString()` equivalent? I don't know the Apple frameworks enough, do you know this by any chance?

If there is built-in thing from Apple, then that message is indeed much better than before, thanks!

Reply:
Thanks for checking! I don't have a Mac, so for me it takes forever to test these little details 😄 

I guess if you find something built-in that has this information, it would be better to use that in case some tooling parses this kind of "logs" in a specific way